### PR TITLE
VDT IT - Add SUSE Linux support to test_scan_results suite

### LINKED ISSUE #2809: Add test_scan_results cases for SUSE Linux Enterprise (1/2)
| Related problem |
| --- |
| https://github.com/wazuh/wazuh-qa/issues/2792 |

### Description

As part of https://github.com/wazuh/wazuh-qa/issues/2792, it is necessary to add cases in the test_scan_results suite related to SUSE Linux Enterprise support for the Wazuh Vulnerability Detector module.

**Suite test_scan_results**

- `Test scan provider and NVD vulnerabilities`: Test that vulnerable packages are reported using the OVAL and NVD feed.
- `Test scan provider vulnerabilities`: Test that vulnerable packages are reported using the OVAL feed (NVD contains different vulnerabilities).
- `Test scan vulnerability removal`: Check that it generates vulnerability alert fixed after removing or updating a vulnerable package.

### To Do 

- [x] Add new custom SUSE feed
- [x] Add scan provider vulnerabilities tests.
- [ ] Add scan provider and NVD vulnerabilities tests
- [ ] Add scan vulnerability removal tests.
- [x] Add SUSE OS to the test documentation

### LINKED ISSUE #2899: Add test_scan_results cases for SUSE Linux Enterprise (2/2)
| Related problem |
| --- |
| https://github.com/wazuh/wazuh-qa/issues/2792 |

### Description

As part of https://github.com/wazuh/wazuh-qa/issues/2792, it is necessary to add cases in the test_scan_results suite related to SUSE Linux Enterprise support for the Wazuh Vulnerability Detector module.

**Suite test_scan_results**

- `Test scan provider and NVD vulnerabilities`: Test that vulnerable packages are reported using the OVAL and NVD feed.
- `Test scan provider vulnerabilities`: Test that vulnerable packages are reported using the OVAL feed (NVD contains different vulnerabilities).
- `Test scan vulnerability removal`: Check that it generates vulnerability alert fixed after removing or updating a vulnerable package.

### To Do 

- [x] Add scan provider and NVD vulnerabilities tests
- [x] Add scan vulnerability removal tests.

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|---|
| close #2809 |
| close #2899 |

## Description
add cases in the test_scan_results suite related to SUSE Linux Server and Desktop support for the Wazuh Vulnerability Detector module.

**Suite test_scan_results**

- `Test scan no agent data`: Check an alert message is shown when attempting scan on Linux agents with NVD feed only
- `Test scan provider vulnerabilities`: Check if the vulnerability alerts are reported making vulnerabilities found only by OVAL and not NVD
- `Test scan provider and nvd vulnerabilities`: Check if the vulnerability alerts are reported making vulnerabilities found by OVAL and NVD
- `Test scan vulnerability removal`: Test that it is reported when a vulnerability is deleted or a package is updated

## Configuration options

local_internal_options
> wazuh_modules.debug=2
monitord.rotate_log=0

The test will be run when the development was finished
## Tests Results
| Test Path | Os/Type | Local | Jenkins | Date | By |
|--|--|--|--|--|--|
|  test/integration/test_vulnerability_detector/test_scan_results | CentOS - Manager | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8708036/R1.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8708037/R2.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8708038/R3.zip)  | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/26421/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/26424/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/26425/)| 2022/05/17  |  @Deblintrake09  | 

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

File: tests/integration/test_vulnerability_detector/data/feeds/suse/custom_suse_oval_feed.xml [hunk 1]
```diff
@@ -8,185 +8,218 @@
...
-    <rpminfo_object id="oval:org.opensuse.security:obj:2009047546" version="1" xmlns="http://oval.mitre.org/XMLSchema/oval-definitions-5#linux">
-		<name>SLES_SAP-release</name>
+		<rpminfo_object id="oval:org.opensuse.security:obj:6" version="1" xmlns="http://oval.mitre.org/XMLSchema/oval-definitions-5#linux">
+		<name>sled-release</name>
 	</rpminfo_object>
-    <rpminfo_object id="oval:org.opensuse.security:obj:2009039405" version="1" xmlns="http://oval.mitre.org/XMLSchema/oval-definitions-5#linux">
-		<name>sle-we-release</name>
+		<rpminfo_object id="oval:org.opensuse.security:obj:7" version="1" xmlns="http://oval.mitre.org/XMLSchema/oval-definitions-5#linux">
+		<name>sles-release</name>
 	</rpminfo_object>
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/data/feeds/suse/custom_suse_oval_feed.xml]: Fixed

File: tests/integration/test_vulnerability_detector/data/feeds/suse/custom_suse_oval_feed.xml [hunk 2]
```diff
@@ -8,185 +8,218 @@
 	  <generator>
 	      <oval:product_name>Marcus Updateinfo to OVAL Converter</oval:product_name>
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/data/feeds/suse/custom_suse_oval_feed.xml]: This was the name on the base Feed i worked on. Fixed

File: deps/wazuh_testing/wazuh_testing/modules/vulnerability_detector/event_monitor.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def check_vuln_detector_event(file_monitor=None, callback='', error_message=None
         error_message
 
     file_monitor.start(timeout=timeout, update_position=update_position, accum_results=accum_results,
-                      callback=make_vuln_callback(callback, prefix), error_message=error_message)
+                       callback=make_vuln_callback(callback, prefix), error_message=error_message)
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/modules/vulnerability_detector/event_monitor.py:55]: It is correctly indented inside the file

File: deps/wazuh_testing/wazuh_testing/mocking/__init__.py [hunk 4]
```diff
@@ -103,7 +103,26 @@
...
+               'os_platform': 'sles', 'name': 'Server12', 'os_codename':'sles'},
+    'SLES15': {'hostname': 'localhost', 'architecture': 'x64', 'os_name': 'SLES', 'os_version': '15.2',
+                 'os_codename': '', 'os_major': '15', 'os_minor': '', 'os_patch': '', 'os_build': '',
+                 'os_platform': 'sles', 'sysname': 'Linux', 'release': '5.3.18-22-default', 'version': 'Wazuh v4.4.0',
+                 'os_release': '', 'checksum': '1652388661375945607', 'name': 'SUSE15', 'ip': '127.0.0.1',
+                 'register_ip': 'any', 'internal_key': '',
+                 'os_uname': 'Linux |localhost|5.3.18-22-default |#1 SMP Wed Jun 3 12:16:43 UTC 2020 (720aeba)|x86_64',
+                 'os_arch': 'x64', 'config_sum': '', 'merged_sum': '', 'manager_host': 'localhost.localdomain',
+                 'node_name': 'node01', 'date_add': '1652381429', 'last_keepalive': '253402300799',
+                 'sync_status': 'synced', 'connection_status': 'active'}
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/mocking/__init__.py]: fixed. The spaces are complying to PEP8. there were spaces missing.

File: deps/wazuh_testing/wazuh_testing/modules/vulnerability_detector/__init__.py [hunk 5]
```diff
@@ -130,6 +141,24 @@ def insert_vulnerabilities_agent_inventory(agent_id='000', status='VALID'):
...
+
+    Args:
+        agent_id (str): Agent ID.
+        version (str): Package version.
+
+    Raises:
+        ValueError: If version parameter has an invalid value.
+    """
+    if version not in SUSE_SYSTEM_PACKAGE:
+        raise ValueError(f"Suse System parameter invalid.")
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/modules/vulnerability_detector/__init__.py]: Fixed

File: deps/wazuh_testing/wazuh_testing/modules/vulnerability_detector/__init__.py [hunk 6]
```diff
@@ -130,6 +141,24 @@ def insert_vulnerabilities_agent_inventory(agent_id='000', status='VALID'):
                                                          cve=package['cveid'], status=status)
 
 
+def insert_suse_system_package(agent_id='000', version='SLES15'):
+    """Insert suse os package to an agent.
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/modules/vulnerability_detector/__init__.py]: Fixed

File: tests/integration/test_vulnerability_detector/test_scan_results/data/configuration_template/configuration_scan_provider_and_nvd_vulnerabilities.yaml [hunk 7]
```diff
@@ -244,3 +244,31 @@
...
+                    - path: CUSTOM_SUSE_OVAL_FEED
+                  value: '15-server'
+        - provider:
+            attributes:
+              - name: 'nvd'
+            elements:
+              - enabled:
+                  value: 'yes'
+              - path:
+                  value: CUSTOM_NVD_JSON_FEED
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/configuration_template/configuration_scan_provider_and_nvd_vulnerabilities.yaml]: Fixed

File: tests/integration/test_vulnerability_detector/test_scan_results/data/configuration_template/configuration_scan_provider_vulnerabilities.yaml [hunk 8]
```diff
@@ -244,3 +244,30 @@
...
+                    - path: CUSTOM_SUSE_OVAL_FEED
+                  value: '15-server'
+        - provider:
+            attributes:
+              - name: 'nvd'
+            elements:
+              - enabled:
+                  value: 'yes'
+              - path:
+                  value: CUSTOM_NVD_JSON_FEED
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/configuration_template/configuration_scan_provider_vulnerabilities.yaml]: Fixed

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_provider_and_nvd_vulnerabilities.yaml [hunk 9]
```diff
@@ -59,3 +59,15 @@
...
+  description: 'Scan SLES vulnerabilities using provider and NVD feed'
+  configuration_parameters: null
+  metadata:
+    provider_name: 'suse'
+    system: 'SLES15'
+    json_feed: null
+    oval_feed: 'custom_suse_oval_feed.xml'
+    nvd_feed: 'custom_nvd_feed.json'
+    oval_feed_tag: CUSTOM_SUSE_OVAL_FEED
+    nvd_feed_tag: CUSTOM_NVD_JSON_FEED
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_provider_and_nvd_vulnerabilities.yaml]: Fixed

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_provider_vulnerabilities.yaml [hunk 10]
```diff
@@ -59,3 +59,15 @@
...
+  description: 'Scan SLES vulnerabilities using provider and NVD feed'
+  configuration_parameters: null
+  metadata:
+    provider_name: 'suse'
+    system: 'SLES15'
+    json_feed: null
+    oval_feed: 'custom_suse_oval_feed.xml'
+    nvd_feed: 'custom_nvd_alternative_feed.json'
+    oval_feed_tag: CUSTOM_SUSE_OVAL_FEED
+    nvd_feed_tag: CUSTOM_NVD_JSON_FEED
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_provider_vulnerabilities.yaml]: Fixed

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_provider_vulnerabilities.py [hunk 11]
```diff
@@ -92,7 +96,7 @@ def test_scan_provider_vulnerabilities(configuration, metadata, agent_system, se
         - Check that the vulnerabilities have been detected for all packages using only OVAL source. (0 for NVD)
         - Check that vulnerability alerts have been generated for all packages.
 
-    wazuh_min_version: 4.3.0
+    wazuh_min_version: 4.4.0
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/test_scan_provider_vulnerabilities.py:99]: Fixed

File: deps/wazuh_testing/wazuh_testing/db_interface/cve_db.py [hunk 12]
```diff
@@ -227,3 +228,35 @@ def get_nvd_metadata_timestamp(year):
...
+        dependency_id (str): id of the dependency
+        name (str): name of the dependency. Defaults to sled-release
+        operation (str): operator used to compare the version
+        operation_value (str): version of the dependency
+        target (str): OS of the pkg
+        installed (str): flag that will be activated in the pre-scan as long as this dependency has been detected in the agent
+        package_id (str): the id to relate the dependency to a package
+    """
+    query_string = f"INSERT OR IGNORE INTO dependencies (id, name, operation, operation_value, target, installed)"\
+                   f"VALUES ('{dependency_id}', '{name}', '{operation}', '{operation_value}', '{target}','{installed}')"
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/db_interface/cve_db.py]: This function was removed becayse it is not being used

File: deps/wazuh_testing/wazuh_testing/db_interface/__init__.py [hunk 13]
```diff
@@ -87,7 +88,7 @@ def execute_sqlite_query(cursor, query):
             if str(exception_message) == 'database is locked':
                 sleep(0.5)
                 retries += 1
-
+
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/db_interface/__init__.py]: Removed

**Review by Deblintrake09 — COMMENTED** [review]
Applied requested changes

File: deps/wazuh_testing/wazuh_testing/db_interface/cve_db.py [hunk 14]
```diff
@@ -227,3 +228,35 @@ def get_nvd_metadata_timestamp(year):
...
+
+    Args:
+        dependency_id (str): id of the dependency
+        target (str): OS of the pkg
+        package_id (str): the id to relate the dependency to a package
+    """
+    pkg_deps_string = f"NSERT INTO PKG_DEPS (pkg_id, dep_id, target) " \
+                      f"VALUES ('{package_id}', '{dependency_id}', '{target}')"
+
+    make_sqlite_query(CVE_DB_PATH, [pkg_deps_string])
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/db_interface/cve_db.py]: Removed

File: tests/integration/test_vulnerability_detector/conftest.py [hunk 15]
```diff
@@ -2,8 +2,11 @@
 # Created by Wazuh, Inc. <info@wazuh.com>.
 # This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
 import pytest
+import os
+from time import sleep
```
Comment (damarisg) [inline tests/integration/test_vulnerability_detector/conftest.py:6]: If it is not used, it should be removed.

File: tests/integration/test_vulnerability_detector/conftest.py [hunk 16]
```diff
@@ -2,8 +2,11 @@
...
 from subprocess import CalledProcessError
 
+from wazuh_testing import CVE_DB_PATH
```
Comment (damarisg) [inline tests/integration/test_vulnerability_detector/conftest.py:9]: If it is not used, it should be removed.

**Review by damarisg — CHANGES_REQUESTED** [review]

**Review by damarisg — APPROVED** [review]
## Tests Results
| Test Path | Os/Type | Jenkins | Date | By |
|--|--|--|--|--|
|  test/integration/test_vulnerability_detector/test_scan_results | CentOS - Manager |  [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/26543/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/26544/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/26545/)| 2022/05/19 | Seyla  |

**Pull request merged**